wg-api/spec-documents/clipboard-rearchitecture.md

```diff
@@ -0,0 +1,105 @@
+2. Alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t addressed by the specification.
+3. Leave API as-is and continue to adapt it as required by upstream changes by Chromium development.
+ 
+### Preferred Approach/Solution
+
+Alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t considered by the specification.
+
+**APIs to Preserve**
+* `clipboard.readBookmark()`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.writeBookmark(title, url[, type])`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.readFindText()` _macOS_
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.writeFindText(text)`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.clear([type])`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.readText([type])`
+  * Already implemented per specification.
+* `clipboard.writeText(text[, type])`
+  * Already implemented per specification.
+
+**APIs to Remove**
+* `clipboard.availableFormats([type])`
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.has(format[, type])` 
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.readBuffer(format)`
+  * Superseded by `clipboard.read`[Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.writeBuffer(format, buffer[, type])`
+  * Superseded by `clipboard.write` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write)
+* `clipboard.readHTML()`
+  * Superseded by `clipboard.read` with `text/html` MIME type
+* `clipboard.writeHTML(markup[, type])`
+  * Superseded by `clipboard.write` with `text/html` MIME type
+* `clipboard.readImage([type])`
+  * Superseded by `clipboard.read()` with `image/*` (where * is bmp, gif, jpeg, png, svg+xml, etc.)
+* `clipboard.writeImage(image[, type])`
+  * Superseded by `clipboard.write`  with `image/*` (where * is bmp, gif, jpeg, png, svg+xml, etc.)
+* `clipboard.readRTF([type])`
+  * Superseded by `clipboard.read` with `application/rtf` MIME type
+* `clipboard.writeRTF(text[, type])`
+  * Superseded by `clipboard.write` with `application/rtf` MIME type
+
+**APIs to Modify**
+* `clipboard.read(format)`
+  * Modify to bring into spec compliance with [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read) `clipboard.read`
+  * Ensure that raw formats are preserved
+    * Custom MIME types e.g. `electron/filepath`
+* `clipboard.write(data[, type])​`
+  * Modify to bring into spec compliance with [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read) `clipboard.read`
+  * Ensure that raw formats are preserved
+    * Custom MIME types e.g. `electron/filepath`
```

Comment:
It looks like there's a muddling of the meaning of 'type' here -- in `main` the type is clipboard | selection; here, it's mime-type.

If this draft was implemented, would Electron lose the ability to read/write from the Linux selection buffer?

Reply:
@ckerr the `type` here is the current argument we have to use the Linux selection buffer.

Reply:
Got it. I was misunderstanding how the W3C spec denoted mime-types.